modules/transactions/src/test/java/org/apache/ignite/internal/tx/impl/TransactionExpirationRegistryTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.tx.impl;
+
+import static org.apache.ignite.internal.util.CompletableFutures.nullCompletedFuture;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.apache.ignite.internal.hlc.HybridTimestamp;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.apache.ignite.internal.tx.InternalTransaction;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class TransactionExpirationRegistryTest extends BaseIgniteAbstractTest {
+    private final TransactionExpirationRegistry registry = new TransactionExpirationRegistry();
+
+    @Mock
+    private InternalTransaction tx1;
+
+    @Mock
+    private InternalTransaction tx2;
+
+    @BeforeEach
+    void configureMocks() {
+        lenient().when(tx1.rollbackAsync()).thenReturn(nullCompletedFuture());
+        lenient().when(tx2.rollbackAsync()).thenReturn(nullCompletedFuture());
+    }
+
+    @Test
+    void abortsTransactionsBeforeExpirationTime() {
+        registry.register(tx1, new HybridTimestamp(1000, 0));
```

Comment:
Why do we use `HybridTimestamp` here? I think it is enough to use just millis.

Reply:
I mean all time-related calculations are done on one coordinator. So, do we really need `HybridTimestamp`?

Reply:
Switched to just physical time (longs)